server/src/main/java/org/opensearch/index/query/AbstractQueryBuilder.java

```diff
@@ -112,6 +113,14 @@ protected void printBoostAndQueryName(XContentBuilder builder) throws IOExceptio
         }
     }
 
+    public String fieldName() {
+        return "none";
```

Comment:
Thoughts on what the default value should be? Some child query builders don't have field name so won't override this method in `AbstractQueryBuilder`.

Reply:
Ideally this fieldName method should be abstract any new implementation of `AbstractQueryBuilder` returns the correct/default fieldName?

Reply:
I agree but there are 20+ existing QueryBuilders which do not already implement `fieldName()`. So if I make it abstract we need to implement the method for these 20+ classes. This can be done, but still need some default/null value.

Example QueryBuilders which do not implement fieldName:
```
/Users/davizane/OpenSearch_dzane17_forks/OpenSearch/server/src/main/java/org/opensearch/index/query/ScriptQueryBuilder.java:67: error: ScriptQueryBuilder is not abstract and does not override abstract method fieldName() in AbstractQueryBuilder
public class ScriptQueryBuilder extends AbstractQueryBuilder<ScriptQueryBuilder> {
       ^
/Users/davizane/OpenSearch_dzane17_forks/OpenSearch/server/src/main/java/org/opensearch/index/query/IntervalQueryBuilder.java:55: error: IntervalQueryBuilder is not abstract and does not override abstract method fieldName() in AbstractQueryBuilder
public class IntervalQueryBuilder extends AbstractQueryBuilder<IntervalQueryBuilder> {
       ^
/Users/davizane/OpenSearch_dzane17_forks/OpenSearch/server/src/main/java/org/opensearch/index/query/TermsSetQueryBuilder.java:70: error: TermsSetQueryBuilder is not abstract and does not override abstract method fieldName() in AbstractQueryBuilder
public final class TermsSetQueryBuilder extends AbstractQueryBuilder<TermsSetQueryBuilder> {
             ^
/Users/davizane/OpenSearch_dzane17_forks/OpenSearch/server/src/main/java/org/opensearch/index/query/MatchNoneQueryBuilder.java:50: error: MatchNoneQueryBuilder is not abstract and does not override abstract method fieldName() in AbstractQueryBuilder
public class MatchNoneQueryBuilder extends AbstractQueryBuilder<MatchNoneQueryBuilder> {
```

Reply:
As discussed by @jainankitk, we can have one abstract method and another method to return the default field name. This will force any new query builder implementations to think about whether they need to implement the `fieldName()`

Reply:
@deshsidd @jainankitk I made `fieldName()` abstract and implemented QB classes in which the method was missing